extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -38,23 +40,62 @@ export const signFreighterTransaction = createAsyncThunk<
 
 export const submitFreighterTransaction = createAsyncThunk<
   Horizon.TransactionResponse,
-  { signedXDR: string; networkUrl: string },
+  { signedXDR: string; networkDetails: NetworkDetails },
   {
     rejectValue: ErrorMessage;
   }
->("submitFreighterTransaction", async ({ signedXDR, networkUrl }, thunkApi) => {
-  try {
-    return await internalSubmitFreighterTransaction({
-      signedXDR,
-      networkUrl,
-    });
-  } catch (e) {
-    return thunkApi.rejectWithValue({
-      errorMessage: e.message || e,
-      response: e.response?.data,
-    });
-  }
-});
+>(
+  "submitFreighterTransaction",
+  async ({ signedXDR, networkDetails }, thunkApi) => {
+    try {
+      return await internalSubmitFreighterTransaction({
+        signedXDR,
+        networkDetails,
+      });
+    } catch (e) {
+      return thunkApi.rejectWithValue({
+        errorMessage: e.message || e,
+        response: e.response?.data,
+      });
+    }
+  },
+);
+
+export const signWithLedger = createAsyncThunk<
+  string,
+  { transactionXDR: string; networkPassphrase: string; publicKey: string },
+  { rejectValue: ErrorMessage }
+>(
+  "signWithLedger",
+  async ({ transactionXDR, networkPassphrase, publicKey }, thunkApi) => {
+    try {
+      const tx = StellarSdk.TransactionBuilder.fromXDR(
+        transactionXDR,
+        networkPassphrase,
+      );
+
+      const transport = await TransportWebUSB.create();
+      const ledgerApi = new LedgerApi(transport);
+      // TODO - move to redux
+      const result = await ledgerApi.signTransaction(
+        "44'/148'/0'",
```

Comment:
might be nice to have a comment explaining the significance of this value

Reply:
it's the default bip path for ledger. Forgot to change this one to the constant, I'll be moving in my next PR but I'll change to the const now so I don't forget 👍 
https://github.com/stellar/freighter/pull/554/commits/7aec9cdaf66534f00ef5374e36bcc38fbad404af